Userland/Libraries/LibJS/Parser.h

```diff
@@ -110,12 +111,28 @@ class Parser {
             // We need to modify the source to match what the lexer considers one line - normalizing
             // line terminators to \n is easier than splitting using all different LT characters.
             String source_string { source };
-            source_string.replace("\r\n", "\n");
-            source_string.replace("\r", "\n");
-            source_string.replace(LINE_SEPARATOR, "\n");
-            source_string.replace(PARAGRAPH_SEPARATOR, "\n");
             StringBuilder builder;
-            builder.append(source_string.split_view('\n', true)[position.value().line - 1]);
+            GenericLexer lexer(source_string);
```

Comment:
Can we remove the String allocation?
```c++
GenericLexer lexer(source);
```

Reply:
Forgot I did that. Master does that because it changes the string, and during my testing, I used the same source_string. Since this should not modify the source_string, it should be safe to do this.